puppet-mode.el

```diff
@@ -632,6 +633,40 @@ of the initial include plus puppet-include-indent."
         (indent-line-to 0)))))
 
 
+;;;; SMIE
+
+(defvar puppet-use-smie nil
+  "Whether to use SMIE or not.")
```

Comment:
The problem is that the current indentation doesn't work very well either. It's better to focus just on one. Perhaps we shouldn't merge this until you get it into some "good enough" state.